distributed/pubsub.py

```diff
@@ -3,20 +3,25 @@
 from contextlib import suppress
 import logging
 import threading
+import typing
```

Comment:
as a personal esthetic preference, I find that using ``from typing import ...`` is always a nicer option. It does lead to collisions with ``from collections.abc import ...``, which can however be worked around after you drop Python 3.6 support, add ``from __future__ import annotations`` everywhere, and you run a Python 3.9-compatible version of mypy (on python 3.7+).

Reply:
I generally agree with you here @crusaderky. In this case I kept it this way because the thing I'm using is `typing.TYPE_CHECKING`, and I didn't really want that floating around the module namespace. No strong preference either way, though.